test/tools/unified-spec-runner/match.ts

```diff
@@ -213,7 +213,18 @@ export function resultCheck(
       }
       resultCheck(actual[key], value, entities, path, checkExtraKeys);
     } else {
-      resultCheck(actual[key], value, entities, path, checkExtraKeys);
+      // If our actual value is a map, such as in the client bulk write results, we need
+      // to convert the expected keys from the string numbers to actual numbers since the key
+      // values in the maps are actual numbers.
+      const isActualMap = actual instanceof Map;
+      const mapKey = !Number.isNaN(key) ? Number(key) : key;
+      resultCheck(
```

Comment:
```suggestion
      const mapKey = !Number.isNaN(Number(key)) ? Number(key) : key;
```

Right? you want to make sure that when converted to a number it doesn't become `NaN` no string will ever return `true` for `Number.isNaN()`

Reply:
Updated.